.github/workflows/merge_to_main.yml

```diff
@@ -0,0 +1,104 @@
+name: Merge to main
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  get-release-tag:
+    name: Publish linux artifacts into s3 bucket
+    runs-on: ubuntu-latest
+    outputs:
+      tag: "tmp-pr-${{ steps.get_pr_number.outputs.result }}"
+    steps:
+      - name: Get PR number
+        uses: actions/github-script@v6
+        id: get_pr_number
+        with:
+          script: |
+            return (
+              await github.rest.repos.listPullRequestsAssociatedWithCommit({
+                commit_sha: context.sha,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+              })
+            ).data[0].number;
+          result-encoding: string
```

Comment:
Don't we get this payload as part of the GH workflow request (`context`) directly? 🤔